tests/test_routing.py

```diff
@@ -83,6 +83,20 @@ def uuid_converter(request):
     return JSONResponse({"uuid": str(uuid_param)})
 
 
+# OData string entity key
+@app.route("/strentity('{param:str}')", name="strentity")
+def strentity(request):
+    text = request.path_params["param"]
+    return JSONResponse({"str": text})
+
+
+# OData integer entity key
+@app.route("/intentity({param:int})", name="intentity")
+def intentity(request):
+    number = request.path_params["param"]
+    return JSONResponse({"int": number})
```

Comment:
Let's just have a single test case, and be more generic in how we phrase it. (I'd never heard of OData URLs)
That'll be enough to test the escaping functionality.

For example, maybe this?...

```python
@app.route("/escaped-brackets({param:int})", name="escaped-brackets")
def escaped_brackets(request):
    """
    A URL route that has some characters that need to be regex-escaped.
    We want this route to match URL strings such as " /escaped-brackets(42)"
    """
    number = request.path_params["param"]
    return JSONResponse({"int": number})
```

Reply:
brackets look like this [ ], and function docstring is not used elsewhere in this module, but comments are.
How about this?
```python
# Route with chars that conflict with regex meta chars
@app.route("/path-with-parentheses({param:int})", name="path-with-parentheses")
def path_with_parentheses(request):
    number = request.path_params["param"]
    return JSONResponse({"int": number})
```

Reply:
I'm fine with just one test case and to remove reference to OData.